### PR TITLE
LicenseInfo: Add spdx parsing: implementation

### PR DESCRIPTION
**Pull request by bs-jokri**
PR of SPDX reader

### CODE REVIEW

**Comment by alex-evo** [comment]
Guys, I'm sorry if this causes much triouble, but I had to move some code around in order to be able to actually access all the parts I needed from `ProjectDatabaseHandler`. Also, I changed the signature of `LicenseInfoParser.getLicenseInfo(AttachmentContent)` to accept `Attachment`.
@maximilianhuber, can you please rebase onto `cf41ada`?

**Comment by maximilianhuber** [comment]
@alex-evo The PR is rebased and contains the refactorings

File: libraries/lib-datahandler/src/main/java/com/siemens/sw360/datahandler/thrift/ThriftClients.java [hunk 1]
```diff
@@ -125,4 +126,7 @@ private static TProtocol makeProtocol(String url, String service) {
         return new VulnerabilityService.Client(makeProtocol(BACKEND_URL, VULNERABILITY_SERVICE_URL));
     }
 
+    public LicenseInfoService.Client makeLicenseInfoClient() {
+        return new LicenseInfoService.Client(makeProtocol(BACKEND_URL, PROJECT_SERVICE_URL));
```
Comment (alex-evo) [inline libraries/lib-datahandler/src/main/java/com/siemens/sw360/datahandler/thrift/ThriftClients.java]: It took me quite a while to figure out why everything builds fine, but `LicenseInfoServiceClient` can't find method `getLicenseInfoFileForProject` in runtime.
You may want to correct that `*SERVICE_URL` argument.

File: backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/LicenseInfoHandler.java [hunk 2]
```diff
@@ -35,30 +53,192 @@
  */
 public class LicenseInfoHandler implements LicenseInfoService.Iface {
 
-    private final AttachmentDatabaseHandler attachmentDatabaseHandler;
+    public static final String LICENSE_INFOS_CONTEXT_PROPERTY = "licenseInfos";
+    public static final String LICENSES_CONTEXT_PROPERTY = "licenses";
+    public static final String LICENSE_INFO_TEMPLATE_FILE = "src/main/resources/licenseInfoFile.vm";
```
Comment (alex-evo) [inline backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/LicenseInfoHandler.java]: it used to be just `"licenseInfoFile.vm"`. why the change?

File: backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/parsers/SPDXParser.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    public static final String FILETYPE_SPDX_EXTERNAL = "SPDX";
+    protected static final String FILETYPE_SPDX_INTERNAL = "RDF/XML";
+    private static final List<String> ACCEPTABLE_ATTACHMENT_FILE_EXTENSIONS = ImmutableList.of(
+            "rdf",
+            "spdx"
+    );
+    protected static final List<String> ACCEPTABLE_ATTACHMENT_CONTENT_TYPES = ImmutableList.of(
+            "application/rdf+xml",
+            "application/xml",
+            "plian/text"
```
Comment (mcjaeger) [inline backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/parsers/SPDXParser.java]: plian -> plain

File: backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/parsers/SPDXParser.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+ * @author: alex.borodin@evosoft.com
+ * @author: maximilian.huber@tngtech.com
+ */
+public class SPDXParser extends LicenseInfoParser {
+    public static final String FILETYPE_SPDX_EXTERNAL = "SPDX";
+    protected static final String FILETYPE_SPDX_INTERNAL = "RDF/XML";
+    private static final List<String> ACCEPTABLE_ATTACHMENT_FILE_EXTENSIONS = ImmutableList.of(
+            "rdf",
+            "spdx"
+    );
```
Comment (mcjaeger) [inline backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/parsers/SPDXParser.java]: is the list of attachment types changed then or do we need the attachment types for this? Or shall we use component license information (XML) for this?

File: backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/LicenseInfoHandler.java (hunk 2 above)
Comment (mcjaeger) [inline backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/LicenseInfoHandler.java]: could we have rather placed this file among the *.properties files instead? I gues that is something to be likely changed for a particular sw360 installation.

Comment (alex-evo) [inline backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/LicenseInfoHandler.java]: @mcjaeger yes, it is. please move your comment there

**Comment by alex-evo** [comment]
I am going to merge this PR and fix the important stuff in the base branch so that it becomes testable with both parser types asap.

File: backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/LicenseInfoHandler.java (hunk 2 above)
Comment (maximilianhuber) [inline backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/LicenseInfoHandler.java]: I am confused, i think the IDE has made this change while refactoring the location of `licenseInfoFile.vm`

File: backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/parsers/SPDXParser.java (hunk 4 above)
Comment (maximilianhuber) [inline backend/src/src-licenseinfo/src/main/java/com/siemens/sw360/licenseinfo/parsers/SPDXParser.java]: I am not sure what you mean, that are the possible file extensions a spdx-document could have. Even `.spdx` might have to be removed, since tag/value is not yet parseable.

**Comment by maximilianhuber** [comment]
@alex-evo: thank you for fixing the issues
